project/src/main/java/com/google/sps/data/UserPrefrences.java

```diff
@@ -0,0 +1,109 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+import com.google.maps.model.LatLng;
+
+import java.util.List;
+
+/**
+ * Represents the prefrences entered by a user
+ */
+@AutoValue
+public abstract class UserPrefrences {
+
+  /**
+   * @return the minimal rating that the user wants to consider,
+   * represented by a number between 1-5.
+   */
+  public abstract float minRating();
+
+  /**
+   * @return the maximal range of prices the user wants to consider,
+   * represented by a number between 0-4.
+   */
+  public abstract int maxPriceLevel();
+
+  /**
+   * @return the coordinates of the location the user wants the food to be delivered to.
+  */
+  public abstract LatLng location();
+
+  /**
+  * @return the cuisine types that the user prefers.
+  */
+  public abstract List<String> cuisineTypes();
+
+  /**
+   * @return a builder that enables to build a new Place object
+   */
+  public static Builder builder() {
+    return new AutoValue_UserPrefrences.Builder();
+  }
+
+
+/**
+   * A builder class for creating UserPrefrences objects.
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+    /**
+     * @param minRating the minimum rating that the user wants to consider,
+     *      represented by a number between 1-5
+     * @return a Place builder that enables to continue building
+     */
+    public abstract Builder setMinRating(float minRating);
+
+    /**
+     * @param maxPriceLevel the maximal range of prices the user wants to consider,
+     *      represented by a number between 0-4
+     * @return a Place builder that enables to continue building
+     */
+    public abstract Builder setMaxPriceLevel(int maxPriceLevel);
+
+    /**
+     * @param location the coordinates of the location the user wants the food to be delivered to
+     * @return a Place builder that enables to continue building
+     */
+    public abstract Builder setLocation(LatLng location);
+
+    /**
+     * @param cuisineTypes the cuisine types that the user prefers
+     * @return a Place builder that enables to continue building
+     */
+    public abstract Builder setCuisineTypes(List<String> cuisineTypes);
+
+    /**
+     * Builds the UserPrefrences object according to the data that was set so far.
+     *
+     * @return the object that was built
+     */
+    abstract UserPrefrences autoBuild();
+
+    /**
+     * Concludes the building of a new UserPrefrences instance.
+     * @return the new instance.
+     * @throws IllegalArgumentException if an input isn't valid (rating / price level)
+     */
+    public UserPrefrences build() throws IllegalArgumentException {
```

Comment:
A point to think about - we validate rating and price level. How about the other fields? Let's say a user didn't enter a location or any cuisines - is this valid? When we later use that information, do we consider that it might be empty? maybe we should add default values, like all cuisines for example...

Reply:
Yes we should totaly think about that.

Regarding the locations we discussed yesteday and I understood we agreed that the prefrences form will always pass a valid location (the user can't submit the form until a valid location is entered, and it is checked by the Google Maps API used for getting the coordinates from the user's entered location). So the location validation already happened when reaching the UserPrefrences building. Do you think we should validate it again here?

Regarding cuisinses I will check the fetcher behavour when an empty textsearch string is passed. Anyway I think we should allow the cuisines list to be empty and this situation needs to be handeled at the fetcher, but it is a good point.

Reply:
When I mentioned the location I was thinking about the onboarding form for example, because as you said the user query location will be valid from the query page. I'm not sure we need to verify it, as long as we understand that the preferences don't have to be completely filled for every use. 
I thought a bit more about the cuisines and I think we should have a default that has all of them - because that's the actual default: if he didn't specify anything, he wants all of them. (BTW in the query form there is already logic that makes sure it isn't empty, but I still think we should do that, we can discuss why f2f if you want).

Reply:
Yes an empty cusines list means we don't restrict what cuisine types to search, so I think it is something that needs to be allowed and handled in the fetcher/ configiration file and not here or in the query form. Anyway IMO it doesn't regard this PR so we can continue this discussion offline.